pkg/operators/oci-handler/oci_test.go

```diff
@@ -0,0 +1,347 @@
+				require.Error(t, err)
+				require.Contains(t, err.Error(), test.expectedErrorMsg)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, test.expectedResult, string(result))
+			}
+		})
+	}
+}
+
+func TestConstructTempConfig(t *testing.T) {
+	tests := []struct {
+		name       string
+		input      string
+		wantErr    bool
+		wantConfig map[string]any
+		wantLength int
+	}{
+		{
+			name:    "Invalid no colon",
+			input:   "invalidannotation",
+			wantErr: true,
+		},
+		{
+			name:    "Invalid no equal sign",
+			input:   "source:invalidannotation",
+			wantErr: true,
+		},
+		{
+			name:    "Valid datasource annotation",
+			input:   "source:foo=bar",
+			wantErr: false,
+			wantConfig: map[string]any{
+				"datasources": map[string]any{
+					"source": map[string]any{
+						"annotations": map[string]any{
+							"foo": "bar",
+						},
+					},
+				},
+			},
+			wantLength: 1,
+		},
+		{
+			name:    "Valid field annotation",
+			input:   "source.field:foo=bar",
+			wantErr: false,
+			wantConfig: map[string]any{
+				"datasources": map[string]any{
+					"source": map[string]any{
+						"fields": map[string]any{
+							"field": map[string]any{
+								"annotations": map[string]any{
+									"foo": "bar",
+								},
+							},
+						},
+					},
+				},
+			},
+			wantLength: 2,
+		},
+		// TODO: Shouldn't these be invalid?
+		{
+			name:    "Invalid multiple colons",
+			input:   "source:field:foo=bar",
+			wantErr: true,
+		},
+		{
+			name:    "Invalid multiple equal signs",
+			input:   "source:foo=bar=baz",
+			wantErr: true,
+		},
+		{
+			name:    "Invalid multiple dots",
+			input:   "source.field.foo:foo=bar",
+			wantErr: true,
+		},
+		{
+			name:    "Empty annotation key",
+			input:   "source:=bar",
+			wantErr: false,
+			wantConfig: map[string]any{
+				"datasources": map[string]any{
+					"source": map[string]any{
+						"annotations": map[string]any{
+							"": "bar",
+						},
+					},
+				},
+			},
+			wantLength: 1,
+		},
+		{
+			name:    "Empty annotation value",
+			input:   "source:foo=",
+			wantErr: false,
+			wantConfig: map[string]any{
+				"datasources": map[string]any{
+					"source": map[string]any{
+						"annotations": map[string]any{
+							"foo": "",
+						},
+					},
+				},
+			},
+			wantLength: 1,
+		},
+		{
+			name:    "Empty subject",
+			input:   ":foo=bar",
+			wantErr: false,
+			wantConfig: map[string]any{
+				"datasources": map[string]any{
+					"": map[string]any{
+						"annotations": map[string]any{
+							"foo": "bar",
+						},
+					},
+				},
+			},
+			wantLength: 1,
+		},
+		{
+			name:    "Empty field name",
+			input:   "source.:foo=bar",
+			wantErr: false,
+			wantConfig: map[string]any{
+				"datasources": map[string]any{
+					"source": map[string]any{
+						"fields": map[string]any{
+							"": map[string]any{
+								"annotations": map[string]any{
+									"foo": "bar",
+								},
+							},
+						},
+					},
+				},
+			},
+			wantLength: 2,
```

Comment:
I think for the 3 test cases I adjusted we also want to return an error for them. Could you fix the logic in `oci.go` for them in a separate commit?
cc @flyth 
```suggestion
		{
			name:    "Empty annotation key",
			input:   "source:=bar",
			wantErr: true,
		},
		{
			name:    "Empty annotation value",
			input:   "source:foo=",
			wantErr: false,
			wantConfig: map[string]any{
				"datasources": map[string]any{
					"source": map[string]any{
						"annotations": map[string]any{
							"foo": "",
						},
					},
				},
			},
			wantLength: 1,
		},
		{
			name:    "Empty subject",
			input:   ":foo=bar",
			wantErr: true,
		},
		{
			name:    "Empty field name",
			input:   "source.:foo=bar",
			wantErr: true,
```